d3/hyper.js

```diff
@@ -85,16 +86,35 @@ function convertHyper(node, parentKey) {
             if (!Array.isArray(innerNodes)) {
                 innerNodes = [innerNodes];
             }
-            children.push({tag: key, children: innerNodes});
+            additionalChildren.push({tag: key, children: innerNodes});
             return;
         });
         // Display the cardinality on the links between the nodes
         var edgeLabel = node.hasOwnProperty("cardinality") ? common.formatMetric(node.cardinality) : undefined;
+        // Collapse nodes as appropriate
+        var children;
+        var _children;
+        if (node.hasOwnProperty("plan")) {
+            // The top-level plan element needs special attention: we want to hide the `header` by default
+            _children = explicitChildren.concat(additionalChildren);
+            var planIdx = _children.findIndex(function(n) {
+                return n.tag === "plan";
+            });
+            children = [_children[planIdx]];
+        } else if (node.hasOwnProperty("operator")) {
+            // For operators, the additionalChildren are collapsed by default
+            children = explicitChildren;
+            _children = explicitChildren.concat(additionalChildren);
+        } else {
+            // Everything else (usually expressions): display uncollapsed
+            children = explicitChildren.concat(additionalChildren);
+        }
         // Build the converted node
         var convertedNode = {
             tag: tag,
             properties: properties,
             children: children,
+            _children: _children,
```

Comment:
what is the variable `_children` for?

Reply:
`children` and `_children` are used for collapsing. Each click which expands/collapses a node internally only swaps `children` and `_children`. This means, `children` should contain the initially visible children while `_children` should contain all children, including the ones which are hidden by default